lib/watchman.ex

```diff
@@ -14,28 +14,49 @@ defmodule Watchman do
     Supervisor.start_link(children, opts)
   end
 
-  def submit(name, value, type \\ :gauge) do
-    Watchman.Server.submit(name, value, false, type)
+  def submit(name, value, type \\ :gauge)
+
+  def submit({external, name}, value, type) when external in [:internal, :external, :always] do
+    Watchman.Server.submit([{external, name}], value, type)
+  end
+
+  def submit(name_list, value, type) when is_list(name_list) do
+    Watchman.Server.submit(name_list, value, type)
+  end
+
+  def submit(name, value, type) do
+    Watchman.Server.submit([{:internal, name}], value, type)
+  end
+
+  def increment(name_list) when is_list(name_list) do
+    submit(name_list, 1, :count)
   end
 
-  def submit(name, value, external, type) do
-    Watchman.Server.submit(name, value, external, type)
+  def increment({external, name}) when external in [:internal, :external, :always] do
```

Comment:
`increment`, `decrement`, and `benchmark` are using `submit`, so maybe we do not need to duplicate the `name` logic in them?